sagemaker-python-sdk/pytorch_horovod_mpi/pytorch_horovod_mpi.ipynb

```diff
@@ -0,0 +1,791 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "dataloader = torch.utils.data.DataLoader(dataset, batch_size=4)\n",
+    "images, labels = next(iter(dataloader))\n",
+    "grid = torchvision.utils.make_grid(images)\n",
+    "plt.imshow(grid.permute(1, 2, 0))  \n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Uploading the Data\n",
+    "\n",
+    "We are going to use the `sagemaker.Session.upload_data` function to upload our datasets to an S3 location. The return value inputs identifies the location -- we will use later when we start the training job.\n",
+    "\n",
+    "To upload our data to S3, we need to specify an S3 bucket and prefix. These should be within the same region as the Notebook Instance, training, and hosting."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "'upload_data' method will be deprecated in favor of 'S3Uploader' class (https://sagemaker.readthedocs.io/en/stable/s3.html#sagemaker.s3.S3Uploader) in SageMaker Python SDK v2.\n"
+     ]
+    }
+   ],
+   "source": [
+    "import sagemaker\n",
+    "from sagemaker.s3 import S3Uploader\n",
+    "\n",
+    "sagemaker_session = sagemaker.Session()\n",
+    "bucket = sagemaker_session.default_bucket()\n",
+    "prefix = 'pytorch-horovod-mpi-example'\n",
+    "inputs = S3Uploader.upload('data', 's3://{}/{}/data'.format(bucket, prefix))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Training\n",
+    "\n",
+    "In order to use Horovod with OpenMPI, we'll have to use a shell script as the training entry point. The shell script invoke our Python training script as part of an Open MPI command.\n",
```

Comment:
I think it'd make more sense to link to https://horovod.readthedocs.io/en/stable/mpirun.html here (rather than in the intro)